kvbc/include/categorization/kv_blockchain.h

```diff
@@ -90,6 +102,7 @@ class KeyValueBlockchain {
   // Get the updates that were used to create `block_id`.
   std::optional<Updates> getBlockUpdates(BlockId block_id) const;
 
+  std::unordered_map<std::string, std::vector<std::string>> getBlockStaleKeys(BlockId block_id);
```

Comment:
Can be made const.

Also, maybe we can put a comment to explain what the output map contains (i.e. category ID -> stale keys).

Finally, maybe we can use `std::map` for uniformity and deterministic outputs between invocations. I think it makes sense to do that.